### DISCUSSION TITLE
CS2-Ultimate-Mapchooser - !mapcycle

### DISCUSSION THREAD

**Discussion by 4fc7c8f789** [kus/cs2-modded-server#70] [General]
Hey!

For CS2-Ultimate-Mapchooser, is there any chance that you could add a !mapcycle command that prints the full css_mapgroup array to chat (PrintToChat). 

Are there any other options that come to your mind how I can achieve this with limited C# / CSS dev knowledge?

**Comment by ALegitCookie** [comment]
i'm not sure if one exists for CS2 yet, but I believe a map rotation plugin similar to RockTheVote is what you're looking for. Unless Kus has one built into the setup already. In which case yes that would be a good feature for him to look into.

**Reply by 4fc7c8f789** [reply]
Yes kus has a non-public plugin implement called 'CS2-Ultimate-Mapchooser', which I am using. It cycles throug a predetermined list of maps (no rtv / nominate function yet). 

Based on what I've found online it is unique in that it leverages mapgroups for the mapcycle rather then hardcoded .txt files. That allows you to dynamically switch the mapcycle as you go from one gamemode to another etc....

Let me know in case you are aware of any other solutions that could be interesting!
